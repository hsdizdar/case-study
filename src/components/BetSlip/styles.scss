.slip-container {
    width: 400px;
    height: auto;
    background-color: #a09fa1;
    border: 2px solid #585758;
    position: fixed;
    bottom: 0;
    right: 0;
    padding: 5px;

    &__match-item {
        display: flex;
        flex-direction: row;
        border-bottom: 1px solid;
        padding: 5px 10px;
    }

    &__amount {
        font-weight: bold;
        padding-top: 5px;
    }
}
