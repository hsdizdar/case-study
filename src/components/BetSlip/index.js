import React, { useContext } from 'react';

import MatchContext from '../../context/MatchContext';

import './styles.scss';

const BetSlip = () => {
  const context = useContext(MatchContext);

  const matches = context && context.matches; // Destructing because of snapshot testing

  const totalAmount = matches && matches.length
    && matches.map((match) => parseFloat(match.rate)).reduce((a, b) => a * b);

  return (
    <div className="slip-container">
      {matches && matches.map((match) => (
        <div className="slip-container__match-item" key={match.id}>
          <p className="slip-container__text">
            {match.mbs}
            &nbsp;&nbsp;
            {`Kod: ${match.id}`}
            &nbsp;&nbsp;
            {`Ma√ß: ${match.name}`}
            &nbsp;&nbsp;
            {`Oran: ${match.rate}`}
            &nbsp;&nbsp;
            {`MBS: ${match.mbs}`}
          </p>
        </div>
      ))}
      <p className="slip-container__amount">{`Toplam Tutar: ${totalAmount || 0}`}</p>
    </div>
  );
};

export default BetSlip;
