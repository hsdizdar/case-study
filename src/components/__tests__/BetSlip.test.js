import React from 'react';
import renderer from 'react-test-renderer';
import BetSlip from '../BetSlip';

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<BetSlip />).toJSON();
    expect(tree).toMatchSnapshot();
});
