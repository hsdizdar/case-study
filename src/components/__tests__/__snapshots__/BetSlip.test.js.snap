// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="slip-container"
>
  <p
    className="slip-container__amount"
  >
    Toplam Tutar: 0
  </p>
</div>
`;
